File: virtualdemo6.cpp
#include<iostream>
using namespace std;

#pragma pack(1)
class base 
{

    public:
    int i,j;
    void fun()
    { cout<<"base fun\n"; }   //1000

    void gun()
    { cout<<"base gun\n"; }   //2000

    virtual void  sun()
    { cout<<"base sun\n"; }    //3000


    virtual void  run()          //4000
    { cout<<"base run\n"; }

}; //16 bytes

#pragma pack(1)

class derived : public base
{
    public:
    int x;
    void fun()                      //address 5000
    { cout<<"derived fun\n"; }

    void  sun()                     //6000
    { cout<<"derived sun\n"; }

    virtual void  mun()             //7000
    { cout<<"derivrd mun\n"; }

     void  bun()                    //8000
    { cout<<"derived  bun\n"; }



};//20 bytes

int main()
{

    base *bp = new derived();

    
     
    bp->fun();
    bp->gun();
     bp->sun();
    bp->run();
     //bp->mun(); //error
    //bp->bun(); //error

    


    return 0;
}

File: java programming/virtualdemo6.java

class base 
{

    
    int i,j;
    void fun()
    { System.out.println("base fun\n"); }   //1000

    void gun()
    { System.out.println("base gun\n"); }   //2000

     void  sun()
    {System.out.println("base sun\n"); }    //3000


     void  run()          //4000
    { System.out.println("base run\n"); }

} //16 bytes


class derived extends base
{
    
    int x;
    void fun()                      //address 5000
    { System.out.println("derived fun\n"); }

    void  sun()                     //6000
    { System.out.println("derived sun\n"); }

    void  mun()             //7000
    { System.out.println("derivrd mun\n"); }

     void  bun()                    //8000
    { System.out.println("derived  bun\n"); }



}//20 bytes


public class virtualdemo6
 {

    public static void main(String[] args)
     {

        base bp = new derived();

        
        
        bp.fun();
        bp.gun();
        bp.sun();
        bp.run();
        //bp.mun(); //error
        //bp.bun(); //error

        


        
    }
}